spring-cloud-sleuth-zipkin/src/main/java/org/springframework/cloud/sleuth/zipkin/RestTemplateSender.java

```diff
@@ -0,0 +1,74 @@
+package org.springframework.cloud.sleuth.zipkin;
+
+import java.net.URI;
+import java.util.List;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.RequestEntity;
+import org.springframework.web.client.RestTemplate;
+
+import zipkin.reporter.BytesMessageEncoder;
+import zipkin.reporter.Callback;
+import zipkin.reporter.Encoding;
+import zipkin.reporter.Sender;
+
+final class RestTemplateSender implements Sender {
+	final RestTemplate restTemplate;
+	final String url;
+
+	RestTemplateSender(RestTemplate restTemplate, String baseUrl) {
+		this.restTemplate = restTemplate;
+		this.url = baseUrl + (baseUrl.endsWith("/") ? "" : "/") + "api/v1/spans";
+	}
+
+	@Override public Encoding encoding() {
+		return Encoding.JSON;
+	}
+
+	@Override public int messageMaxBytes() {
+		return 5 * 1024 * 1024;
+	}
+
+	@Override public int messageSizeInBytes(List<byte[]> spans) {
+		return encoding().listSizeInBytes(spans);
+	}
+
+	/** close is typically called from a different thread */
+	transient boolean closeCalled;
+
+	@Override public void sendSpans(List<byte[]> encodedSpans, Callback callback) {
```

Comment:
AsyncReporter calls this.. you can't see it here. https://github.com/openzipkin/zipkin-reporter-java/blob/master/core/src/main/java/zipkin/reporter/AsyncReporter.java#L222